pkg/server/server.go

```diff
@@ -0,0 +1,54 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	pb "github.com/rexposadas/teleport/api"
+	"log"
+)
+
+type Server struct {
+	pb.UnimplementedTeleportServiceServer
+
+	jobmanager *JobManager
+}
+
+func NewServer() *Server {
+	return &Server{
+		jobmanager: NewJobManager(),
+	}
+}
+
+func (s *Server) Start(_ context.Context, in *pb.StartRequest) (*pb.StartResponse, error) {
+	job, err := s.jobmanager.StartJob(in.Command, in.Args)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
```

Comment:
To me, it feels like managing the goroutines watching the jobs would be better as the responsibility of the jobmanager rather than the grpc handler.